agents/pfeiffer_tpg366/pfeiffer_tpg366_agent.py

```diff
@@ -0,0 +1,179 @@
+
+        Args:
+            None
+        Returns:
+            6 element array corresponding to each channels pressure reading, as floats
+        """
+        msg = 'PRX\r\n'
+        self.comm.send(msg.encode())
+        status = self.comm.recv(BUFF_SIZE).decode()#Could probably use this to catch exemptions
+        self.comm.send(ENQ.encode())
+        read_str = self.comm.recv(BUFF_SIZE).decode()
+        pressure_str = read_str.split('\r')[0]
+        gauge_states = pressure_str.split(',')[::2]
+        gauge_states = np.array(gauge_states, dtype=int)#this is recorded just incase
+        pressures = pressure_str.split(',')[1::2]
+        pressures = [float(p) for p in pressures]
+        #print(pressures,type(pressures[0]),"This works")
+        return pressures 
+
+    def close(self):
+        """
+        Close the socket of the connection
+        """
+        self.comm.close()
+
+class pfeifferAgent:
+    def __init__(self, agent,ip_address = IP_ADDRESS, port=8000, f_sample=2.5):
+        self.active = True  
+        self.agent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+        self.f_sample = f_sample
+        self.take_data = False
+        self.module: Optional[Module] = None
+        self.gauge = pfeiffer()
+        self.port = port
+        self.ip_address = ip_address
+       
+        agg_params = {'frame length':60,}
+
+        self.agent.register_feed('pressures',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=1)
+    
+    def start_acq(self,session, params = None):
+        """
+        Get pressures from the Pfeiffer gauges, publishes them to the feed
+        
+        Args: 
+            sampling_frequency- defaults to 2.5 Hz
+
+        """
+        if params is None:
+            params = {}
+
+        f_sample = params.get('sampling_frequency')
+        if f_sample is None:
+            f_sample = self.f_sample
+
+        sleep_time = 1./f_sample - 0.01
+
+        with self.lock.acquire_timeout(timeout=0, job='init') as acquired:
+        # Locking mechanism stops code from proceeding if no lock acquired
+            if not acquired:
+                self.log.warn("Could not start init because {} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('running')
+
+            self.take_data = True
+
+            while self.take_data:
+                data = {
+                    'timestamp': time.time(),
+                    'block_name': 'pressures',
+                    'data': {}
+                }
+                pressure_array = self.gauge.read_pressure_all()
+                #Loop through all the channels on the device
+                for i in range(len(pressure_array)):
+                    data['data']["pressure_ch"+str(i)]= pressure_array[i]
```

Comment:
Be aware this is going to index on 0. This is contrary to the convention Bluefors has taken on their vacuum gauge setup.

Reply:
Are you referring to the name of the channels published? For example, we want to see "Channels 1-6" rather than "Channels 0-5".

Reply:
Yes. Maybe you want to see 0-5, but I wanted to point out the difference since the Bluefors system uses 1-6 on what I believe is an identical gauge system.